src/sql/workbench/contrib/notebook/browser/cellViews/interfaces.ts

```diff
@@ -28,7 +28,7 @@ export abstract class CellView extends AngularDisposable implements OnDestroy, I
 
 	public abstract cellGuid(): string;
 
-	public deltaDecorations(newDecorationRange: NotebookRange, oldDecorationRange: NotebookRange): void {
+	public deltaDecorations(newDecorationsRange: NotebookRange | NotebookRange[], oldDecorationsRange: NotebookRange | NotebookRange[]): void {
```

Comment:
Can we just make this only take in an array for simplicity? Callers can just include the single item in an array if needed.

Reply:
the reason I had it this way is, am using the same function call for highlighting all finds and the specific range with current match. If I make it into an array, it wouldn't work where the number of matches is only one. Am using that differentiate between adding highlight all and highlight specific.

Reply:
My suggestion was to make it so that if the number of matches is only one that you pass it in as an array of length one in that case. Is that not something you can easily change?

Reply:
so if there is a cell with only one match and I need to highlight it in yellow when it isn't the current highlight and in orange if is the current one. I pass it in as array for the first scenario and other way for the second scenario. I don't have any other way to differentiate and didn't want to add another parameter for this sake.

Reply:
Why is that the case only for a single match? Highlighting should be the same regardless of the number of matches - the currently selected highlight would be orange and all others (if any) yellow. So whatever logic is being done when there's multiple matches should be the same as when there's one match.

Reply:
So the reason is, we don't actually have the decoration id for markdown text cells but use the ranges and mark library for highlighting the search term.

Reply:
I'm missing some context here I think on how this is all tied together. How does what you said change what I brought up - you still have to highlight differently in both cases right? 

Could you point me to the place where this is being called with a single item?

Reply:
they are called from notebookEditor -> updateDecorations which is called in notebookFindDecorations by addDecorations, clearDecorations, setCurrentFindMatch and removeLastDecoration.